### ISSUE TITLE
[Bug] Disentangler of mortals not working after moving it to a new location.

### ISSUE THREAD

**Issue by macks2008** [Ellpeck/NaturesAura#343]
The Disentangler of Mortals was working fine when I had it next to my Natural Altar, filling a good chunk of the aura bar each time I killed a chicken near it. Then, after moving it to a pen I created using a Corporeal Eye in combination with Powder of Fertility (I figured I'd be able to contain my animals with the eye, use the powder to keep the population up, and use the stable population and some slaughtering to keep the aura up), it stopped working. Currently, whenever I kill a chicken near it, it makes light blue particles, does nothing to the aura, and still destroys the animal's drops. Based on the particles and destroyed drops, it's clear that it registered the animal's death, but then didn't do anything, and the only thing the book says about kills being wasted is if it's already processing a previous kill (which is not the case. It's been minutes and there are no orange particles). The aura also has plenty of capacity.
I've tried the following things to troubleshoot the issue:
- killing chickens in different positions relative the disentangler
- summoning chickens (with the time_live NBT set to 72000 for good measure) and killing them near the disentangler after dismantling the Corporeal Eye and Powder of Fertility
- Killing a legitimately obtained and really old chicken near the disentangler, with the aforementioned pen still dismantled. (This is in case I overestimated how much the chickens I'd been slaughtering up to that point were worth)
- moving the disentangler back to where it was near my Natural Altar and killing another summoned chicken (after removing some aura from the nearly-full field at that location). At this point it worked, but that defeats the purpose of moving the device, and doesn't actually answer what is wrong with the new location.

In short, I'm at a loss.

Here's a picture of the setup I was trying (again, minus the eye and powder):
![2024-01-10_13 00 44](https://github.com/Ellpeck/NaturesAura/assets/7110033/e4ca8ac7-9466-488d-8c5a-4dbfa1e8756f)

Context/Versions:
this is happening on a server running the [ATM 8 modpack](https://www.curseforge.com/minecraft/modpacks/all-the-mods-8).
Modpack version: 1.0.28
Mod version: 38.2
Forge version: 43.2.14
Minecraft version: 1.19.2

**Comment by macks2008** [comment]
Region boundaries wouldn't mess up anything with the logic right? The chunk this issue is happening at is just north of the nearest region boundary. Could it be misreading the aura as if it's full because of that? Not sure how that would work, but it might be worth a look, especially since I moved the disentangler a few blocks south, across said boundary, and it started working... going to try it in a few others positions on either side of the boundary to see if that's it.

**Comment by macks2008** [comment]
Inconclusive, since it mysteriously started working again in the original location after I drained the aura a bit with the /naaura command.
The aura was by no means full, like at the top of the bar or something, so...

**Comment by macks2008** [comment]
Okay it once again stopped working when I moved the disentangler again. 
So far, the positions where it has worked and not worked are  as follows:
- worked when next to my natural altar (position: exactly -1285, 63, 21 r-3.0.mca)
- worked when put in my animal pen from early game (position: approx -1272, 64, 57 r-3.0.mca)
- Did not work initially (until some unknown trigger fixed it) in any of several positions in the chunk containing -1278, 64, -2 (part of region r -3.-1.mca)
- And now it's not working at -1260 65 11...

Figuring I couldn't make it too much worse, I gave the naaura reset command a try before saving this commentt. Unfortunately, I wasn't sure what the integer argument in the command was for, so I guess I just cleared a radius or diameter of 100 meters, because there's no aura anywhere in my base? At which point it started working again, so at least I know for sure it's something weird with the aura!

**Comment by Ellpeck** [comment]
Hi! It just sounds like you're missing a creational catalyst, which is why the disentangler stops working after a certain aura amount in the area is reached. You can check the topmost chapters in the "creating aura" section (the *italics* ones, mainly) to get more info about this if you haven't already :)

Also keep in mind that the disentangler only produces aura from animals that have lived a certain amount of time, usually between one and two hours, or more to generate even more aura.

**Comment by macks2008** [comment]
Yes I'm aware of the creation all catalyst, but how much or is it supposed to be able to create if the bar is only barely above the middle? If I reset the aura, the bar goes all the way to full before it stops, usually getting high enough for at least  the bone meal effect to start happening, yet if I move the disentangler, even to somewhere relatively neutral, it doesn't generate any noticeable aura

**Comment by Ellpeck** [comment]
As the entry on the environmental eye states: it's not as powerful in detecting the amount of aura around it as most aura-related blocks are. Because aura naturally spreads around the environment to equalize, it may be that in an area of 30-40 blocks, the total amount of aura is quite saturated, whereas in an area of the 15-20 blocks that the environmental eye can detect, it may seem like the aura amount is comparatively low.

Of course, this may also be a bug with the disentangler specifically that i'm unaware of, but i suggest trying to put down the creational catalyst and seeing if that solves all of your issues :)

**Comment by macks2008** [comment]
I think that would be precluded if there are only two points at which the aura is beeing modified, wouldn't it?
After resetting the aura, the Disentangler in the new position stopped working when the bar got all the way full (though I don't see the imbalance effect icons here. I'm pretty sure I've gotten it high enough for imbalance effects with just the disentangler, but maybe as you said that requires the whole area to be more saturated)
![image](https://github.com/Ellpeck/NaturesAura/assets/7110033/29d064aa-5f2c-42df-904e-19249afb59d8)
Vs. here when I put it back at my altar. 
![image](https://github.com/Ellpeck/NaturesAura/assets/7110033/33dfc172-3d79-49f6-bfdc-c944f4f3efe3)

I understand what you're saying about the gauge averaging the input, but this seems extreme considering I started from having reset the aura. Both positions should be able to get to roughly the same saturation in that case, right?

While you reply I will see what happens with the creational catalyst. I'm also wondering if things work better if different types of generators are used, does it keep track of what type of generator was used?

**Comment by Ellpeck** [comment]
Yes, you're right in that this case seems a little extreme. The first aura bar screenshot is exactly what you'd expect: the generator stops working as soon as the aura in the area reaches the saturation that the aura bar can display. However, if you then move it to your altar, and the altar isn't that far away (maybe below 30 or so blocks), the generator will still be able to "see" that the aura in the previous area is already very saturated, because the previous area will be in range. So overall, I can still see this being the issue I described earlier :)

The type of generator does not have an impact on the amount of aura generated, or the effects that get produced!

**Comment by macks2008** [comment]
Oh, I think I get it now, and you were right about the creational catalyst. As you seem to have realized, I had (understandably, I think) assumed the generator would only check its exact location, not the area 10-30 blocks away. Why do you have it check the area, though? Can't you just have it so that it stops when it can't disperse the aura it's producing, or else the total aura produced reaches a certain value? Or am I still misunderstanding how you've modeled this?

**Comment by Ellpeck** [comment]
> also, based on what you've said, is 30 blocks a good spacing, in general, for generators? I can go much further if needed, and I'll probably get at least one (legitimate, as opposed to the one I spawned for testing) creational catalyst. I just don't want to keep running into this issue where I waste perfectly healthy animal souls 😄

Aura naturally spreads out the more there is in an area, so it's generally not immediately necessary to use multiple generators or to spread them out in any sort of predetermined range. Keep in mind that you can also use the aura attraction cart to move aura around manually, though!

> Oh, I think I get it now, and you were right about the creational catalyst. As you seem to have realized, I had (understandably, I think) assumed the generator would only check its exact location, not the area 10-30 blocks away. Why do you have it check the area, though? Can't you just have it so that it stops when it can't disperse the aura it's producing, or else the total aura produced reaches a certain value? Or am I still misunderstanding how you've modeled this?

The model is a lot more complicated than that, and simply checking the exact position would not make aura spread or equalize in a way that would work nicely for gameplay. A key part of how the backend is designed is to make the movement and existence of aura seem natural and to now follow a specific predetermined pattern - even though that's, of course, what it does!

I'm glad you figured it out for your case, though! I'll close this issue for now then :)

**Comment by macks2008** [comment]
Understandable, I just hope no one else makes the same mistake I did.  Part of my reason for moving my generator around is that my base is pretty spread out, but that's a good point about using the aura attraction cart.
I still feel like this could be a little bit more intuitive somehow, but based on what you said I'm not quite sure how. It's obviously important to have some degree of abstraction or "hidden" information to the aura, for the same reason Botania's mana doesn't get measured in exact numbers:  "Given the opportunity, players will optimize the fun out of a game". However, I kind of feel betrayed by the Environmental Ocular (So many chickens died in vain, their spirits becoming part of the background radiation that bla… Wait, wrong game). Can you look into it sometime?

**Comment by Ellpeck** [comment]
The book of natural aura already states that the environmental eye is limited and somewhat inaccurate, however I agree that it might make sense to make a little bit clearer what exactly that means for using it! :)

**Comment by macks2008** [comment]
Yeah you can see how I was thrown off by the lack of specificity. I think it just needs to specify that it's range is limited and things might be affected by aura outside that range.

Hey uh, while I have your attention, what's the population limit supposed to be before Powder of Fertility stops working? Because I had a rather extreme population boom the other day and might therefore have another bug to report.
